dds/DCPS/FilterEvaluator.cpp

```diff
@@ -125,11 +125,21 @@ FilterEvaluator::SerializedForEval::lookup(const char* field) const
     return iter->second;
   }
   Message_Block_Ptr mb (serialized_->duplicate());
-  // TODO(iguessthislldo): Use Encap
   Serializer ser(mb.get(),
     cdr_ ? Encoding::KIND_XCDR1 : Encoding::KIND_UNALIGNED_CDR, swap_);
   if (cdr_) {
-    ser.skip(4); // CDR encapsulation header
+    char buffer[4];
+    ser.buffer_read(buffer, 4, false);
+    EncapsulationHeader encap;
+    ACE_UINT16 kind = (ACE_UINT16(buffer[0]) << 8) | ACE_UINT16(buffer[1]);
+    encap.kind(EncapsulationHeader::Kind(kind));
+    ACE_UINT16 options = (ACE_UINT16(buffer[2]) << 8) | ACE_UINT16(buffer[3]);
+    encap.options(options);
```

Comment:
`EncapsulationHeader` has a deserialization function that you should use instead. Here it being used in `DataReaderImpl_T`:

https://github.com/objectcomputing/OpenDDS/blob/90afd889bd79b881f278444902fddb31c1accc04/dds/DCPS/DataReaderImpl_T.h#L983-L997